python-package/lightgbm/sklearn.py

```diff
@@ -673,6 +673,18 @@ def _more_tags(self) -> Dict[str, Any]:
             },
         }
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        more_tags = self._more_tags()
+        tags.input_tags.allow_nan = more_tags.pop("allow_nan", False)
+        tagged_input_types = more_tags.pop("X_types", [])
+        tags.input_tags.sparse = "sparse" in tagged_input_types
+        tags.target_tags.one_d_labels = "1dlabels" in tagged_input_types
+        tags._xfail_checks = more_tags.pop("_xfail_checks", {})
```

Comment:
```suggestion
        tags.input_tags.allow_nan = more_tags["allow_nan"]
        tags.input_tags.sparse = "sparse" in more_tags["X_types"]
        tags.target_tags.one_d_labels = "1dlabels" in more_tags["X_types"]
        tags._xfail_checks = more_tags["_xfail_checks"]
```

We completely control the result of `_more_tags()` here in `lightgbm`'s codebase, so I don't think this flexibility of using `.pop()` with default values is necessary.

Instead, can you please just use direct `[` access as suggested here? It should be totally safe, and it's simpler.